common/tools/dev-tool/src/commands/samples/publish.ts

```diff
@@ -102,6 +102,28 @@ function createPackageJson(info: SampleGenerationInfo, outputKind: OutputKind):
   };
 }
 
+/**
+ * Determines whether a module specifier is a package dependency.
+ *
+ * A dependency is a module specifier that does not refer to a node builtin and
+ * is not a relative path.
+ *
+ * Absolute path imports are not supported in samples (because the package base
+ * is not fixed relative to the source file).
+ *
+ * @param moduleSpecifier - the string given to `import` or `require`
+ * @returns - true if `moduleSpecifier` should be considered a reference to a
+ * node module dependency
+ */
+function isDependency(moduleSpecifier: string): boolean {
+  if (nodeBuiltins.includes(moduleSpecifier)) return false;
+
+  // This seems like a reasonable test for "is a relative path" as long as
+  // absolute path imports are forbidden.
+  const isRelativePath = /^\.\.?\//.test(moduleSpecifier);
+  return !isRelativePath;
```

Comment:
NIT:
```suggestion
  return !(nodeBuiltins.includes(moduleSpecifier) ||
    // This seems like a reasonable test for "is a relative path" as long as
    // absolute path imports are forbidden.
    /^\.\.?\//.test(moduleSpecifier));
```
Not sure about formatting though :)